bentoml/_internal/frameworks/transformers.py

```diff
@@ -415,127 +452,148 @@ def _save(
+            _config = json.loads(of.read())
+            try:
+                arch = "".join(_config["architectures"])
+                if from_tf:
+                    _model.info.options["model"] = f"TF{arch}"
+                elif from_flax:
+                    _model.info.options["model"] = f"Flax{arch}"
+                else:
+                    _model.info.options["model"] = arch
+            except KeyError:
+                _model.info.options["model"] = ""
+
+        # NOTE: With Tokenizer there are way too many files
+        #  to be included, per frameworks. Thus we will load
+        #  a Tokenizer instance, then save it to path.
+        try:
+            _tokenizer_inst: "ext.TransformersTokenizerType" = AutoTokenizer.from_pretrained(  # type: ignore[reportUnknownMemberType]
                 model_identifier,
-                filename=weight_filename,
-                revision=revision,
-                mirror=mirror,
-            )
-            _download_from_hub(
-                weight_file_url,
-                output_dir=_model.path,
-                proxies=proxies,
-                use_auth_token=use_auth_token,
                 force_download=force_download,
                 resume_download=resume_download,
+                proxies=proxies,
+                revision=revision,
+                subfolder=subfolder,
+                use_fast=use_fast,
+                tokenizer_type=tokenizer_type,
+                trust_remote_code=trust_remote_code,
             )
-
-            # _model.info.options will be None here
-            with open(_model.path_of(CONFIG_NAME), "r", encoding="utf-8") as of:
-                _config = json.loads(of.read())
-                try:
-                    arch = "".join(_config["architectures"])
-                    if from_tf:
-                        _model.info.options["model"] = f"TF{arch}"
-                    elif from_flax:
-                        _model.info.options["model"] = f"Flax{arch}"
-                    else:
-                        _model.info.options["model"] = arch
-                except KeyError:
-                    _model.info.options["model"] = ""
-            _model.info.options["pipeline"] = False
-
-            # NOTE: With Tokenizer there are way too many files
-            #  to be included, per frameworks. Thus we will load
-            #  a Tokenizer instance, then save it to path.
-            try:
-                _tokenizer_inst = AutoTokenizer.from_pretrained(model_identifier)
-                _tokenizer_inst.save_pretrained(_model.path)
-                _model.info.options["tokenizer"] = type(_tokenizer_inst).__name__
-            except ValueError:
-                _model.info.options["tokenizer"] = "na"
+            _ = _tokenizer_inst.save_pretrained(_model.path)  # type: ignore[reportUnknownMemberType]
+            _model.info.options["tokenizer"] = type(_tokenizer_inst).__name__
+        except (ValueError, KeyError):
```

Comment:
Is this something we can refactor to a `if else` check?

Reply:
I'm still looking into this changes from transformers to make it a lot more simple. Had a long demo with @bojiang  this morning about this whole segment and found some bugs when we tried to load the a transformers model back.